### DISCUSSION TITLE
testing data availability, pulling only the laptimes

### DISCUSSION THREAD

**Discussion by smartbackwards** [theOehrly/Fast-F1#313] [Q&A]
hi!
i'm making these graphics which visualise testing sessions. i only deal with lap times - is there an option to not pull telemetry data, and thus, get the data maybe a bit earlier than 30 minutes after the session ends? on that topic, will the data from the testing sessions be available 30 minutes after the afternoon session ends, or the morning session ends?

**Comment by theOehrly** [comment]
1) no, telemetry and lap timing data are available at the same time. So there is no advantage to skipping telemetry data (but you can still skip telemetry data for other reasons, like downloading less data, see the docs for ``Session.load``)

2) data from testing sessions may not be available at all after a session. For the last few seasons, the only option was to record the live timing data stream. You should read the documentation for the live timing client thoroughly if you wish to do that. When recording multiple hours of data streams, there are some important things that you need to take care of. But as every year, things might have changed after the winter break and we will only find that out once testing starts.
